### ISSUE TITLE
Improve error handling in ParseAccessTokenResponse 

### ISSUE THREAD

**Issue by mxa0079** [aarondcoleman/Fitbit.NET#106]
From @WestDiscGolf --

'OAuth2Helper` ParseAccessTokenResponse needs updating - Fitbit exception being thrown - new FitbitException for parsing issues for responses we should be able to handle?

---
Code:

```csharp
 internal static OAuth2AccessToken ParseAccessTokenResponse(string responseString)
        {
            JObject responseObject = JObject.Parse(responseString);

            // Note: if user cancels the auth process Jawbone returns a 200 response, but the JSON payload is way different.
            var error = responseObject["error"];
            if (error != null)
            {
                // TODO: Actually should probably raise an exception here maybe?
                //mxa0079: agree. This is not transparent and makes it hard to debug for consumers of the library
                return null;
            }

            var accessToken = new OAuth2AccessToken();

            var temp_access_token = responseObject["access_token"];
            if (temp_access_token != null) accessToken.Token = temp_access_token.ToString();

            var temp_expires_in = responseObject["expires_in"];
            if (temp_expires_in != null) accessToken.ExpiresIn = Convert.ToInt32(temp_expires_in.ToString());

            var temp_token_type = responseObject["token_type"];
            if (temp_token_type != null) accessToken.TokenType = temp_token_type.ToString();

            var temp_refresh_token = responseObject["refresh_token"];
            if (temp_refresh_token != null) accessToken.RefreshToken = temp_refresh_token.ToString();

            return accessToken;
        }
```

**Comment by WestDiscGolf** [comment]
Looking at this now ...

**Comment by WestDiscGolf** [comment]
See latest PR

**Comment by mxa0079** [comment]
Solved in #121
